Add additional Pushover parameters

dale3h opened the pull request:
**Description:**
Add support for more Pushover parameters: target (device), sound, url, url_title, priority, timestamp

**Related issue (if applicable):** fixes #

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#

**Example entry for `configuration.yaml` (if applicable):**

``` yaml
automation:
  alias: Garage Door is Open
  trigger:
    platform: state
    entity_id: sensor.garage_door
    state: 'open'
  action:
    service: notify.pushover
    data:
      title: Garage Door
      message: The garage door is open.
      target: iPhone,MacBookPro
      data:
        url: https://home-assistant.io/
        url_title: Home Assistant
        priority: 1
        timestamp: 1466024112
        sound: siren
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If code communicates with devices:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16)).
- [ ] New dependencies are only imported inside functions that use them ([example](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51)).
- [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
- [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ ] Tests have been added to verify that the new code works.

dale3h committed:
Add additional Pushover parameters

Add support for more Pushover parameters: target (device), sound, url, url_title, priority, timestamp

balloob commented on homeassistant/components/notify/pushover.py:
```diff
@@ -51,7 +51,21 @@ def send_message(self, message="", **kwargs):
         """Send a message to a user."""
         from pushover import RequestError
 
+        data = kwargs.get(ATTR_DATA)
```
Please don't modify the original data dictionary that is being passed in.

You can supply a second argument to `get` that will be returned if the key doesn't exist, that way you don't have to check for `None` on the next line.

```python
# Make a copy and use empty dict as default value
data = dict(kwargs.get(ATTR_DATA, {}))
```

dale3h replied:
Thank you @balloob, I've committed the fix.

I'll be able to run the local tests in a few days when I figure it all out.

dale3h committed:
Remove data dictionary reference

https://github.com/home-assistant/home-assistant/pull/2309#discussion_r67603127

balloob commented:
Looks good! :dolphin:

dale3h commented:
This appears to break if at least one `data:` key is not provided. I am going to try adding the check for `None` back and see if that fixes it.
